libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -281,15 +324,21 @@ bool AP_Arming::is_armed_and_safety_off() const
 
 uint32_t AP_Arming::get_enabled_checks() const
 {
-    return checks_to_perform;
+    // ignore unknown checks
+    uint32_t check_mask = (uint32_t(Check::CHECK_LAST)-1) & (~1); // remove former ALL bit
+    return (~checks_to_skip) & check_mask;
 }
 
 bool AP_Arming::check_enabled(const AP_Arming::Check check) const
 {
-    if (checks_to_perform & uint32_t(Check::ALL)) {
-        return true;
-    }
-    return (checks_to_perform & uint32_t(check));
+    return !(checks_to_skip & uint32_t(check));
```

Comment:
For whatever reason we prefer comparing with `== 0` on these.

Reply:
Done.